src/Apps/Order/Routes/__tests__/Shipping.jest.tsx

```diff
@@ -1077,6 +1178,215 @@ describe("Shipping", () => {
+            })
+          })
+
+          it("goes back and edits address after verification", async () => {
+            const { env } = renderWithRelay(
+              {
+                CommerceOrder: () =>
+                  UntouchedBuyOrderWithArtsyShippingDomesticFromUS,
+                Me: () => meWithoutAddress,
+              },
+              undefined,
+              relayEnv
+            )
+
+            await fillAddressForm(validAddress)
+            await saveAndContinue()
+
+            await verifyAddressWithSuggestions(
+              env,
+              validAddress,
+              recommendedAddress
+            )
+
+            mockCommitMutation
+              .mockResolvedValueOnce(settingOrderArtaShipmentSuccess)
+              .mockImplementationOnce(relayProps => {
+                relayProps[1].onCompleted(saveAddressSuccess)
+              })
+              .mockResolvedValueOnce(selectShippingQuoteSuccess)
+
+            // Clicking "Back to Edit" allows users to edit the address form
+            // and requires clicking "Save and Continue" to proceed.
+            userEvent.click(screen.getByText("Back to Edit"))
+            await userEvent.click(screen.getByText("Save and Continue"))
+            await flushPromiseQueue()
+
+            expect(mockCommitMutation).toHaveBeenCalledTimes(2)
+
+            let mutationArg = mockCommitMutation.mock.calls[0][0]
+            expect(mutationArg.mutation.default.operation.name).toEqual(
+              "SetShippingMutation"
+            )
+            expect(mutationArg.variables).toEqual({
+              input: {
+                id: "2939023",
+                fulfillmentType: "SHIP_ARTA",
+                addressVerifiedBy: "USER",
+                phoneNumber: validAddress.phoneNumber,
+                shipping: {
+                  ...validAddress,
+                  phoneNumber: "",
+                },
+              },
+            })
+
+            mutationArg = mockCommitMutation.mock.calls[1][0][1]
+            expect(mutationArg.mutation.default.operation.name).toEqual(
+              "CreateUserAddressMutation"
+            )
+            expect(mutationArg.variables).toEqual({
+              input: {
+                attributes: {
+                  ...validAddress,
+                  phoneNumber: "5555937743",
+                },
+              },
+            })
+
+            userEvent.click(screen.getByText(/^Premium/))
+            await saveAndContinue()
+
+            expect(mockCommitMutation).toHaveBeenCalledTimes(4)
+
+            mutationArg = mockCommitMutation.mock.calls[2][0]
+            expect(mutationArg.mutation.default.operation.name).toEqual(
+              "SelectShippingOptionMutation"
+            )
+            expect(mutationArg.variables).toEqual({
+              input: {
+                id: "2939023",
+                selectedShippingQuoteId: "1eb3ba19-643b-4101-b113-2eb4ef7e30b6",
+              },
+            })
```

Comment:
non-blocking: `expect(mockCommitMutation).toHaveBeenNthCalledWith(3, {...})` is also an option here. [Example](https://github.com/artsy/force/blob/4566d16b6a3fea1aff1c3022855a3baed6ed8d2b/src/Apps/Order/Components/__tests__/AddressVerificationFlow.jest.tsx#L329)